iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/region/AddRegionPeerProcedure.java

```diff
@@ -0,0 +1,203 @@
+import java.nio.ByteBuffer;
+
+import static org.apache.iotdb.confignode.procedure.state.AddRegionPeerState.UPDATE_REGION_LOCATION_CACHE;
+import static org.apache.iotdb.rpc.TSStatusCode.SUCCESS_STATUS;
+
+public class AddRegionPeerProcedure
+    extends StateMachineProcedure<ConfigNodeProcedureEnv, AddRegionPeerState> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AddRegionPeerProcedure.class);
+  private TConsensusGroupId consensusGroupId;
+
+  private TDataNodeLocation coordinator;
+
+  private TDataNodeLocation destDataNode;
+
+  public AddRegionPeerProcedure() {
+    super();
+  }
+
+  public AddRegionPeerProcedure(
+      TConsensusGroupId consensusGroupId,
+      TDataNodeLocation coordinator,
+      TDataNodeLocation destDataNode) {
+    super();
+    this.consensusGroupId = consensusGroupId;
+    this.coordinator = coordinator;
+    this.destDataNode = destDataNode;
+  }
+
+  @Override
+  protected Flow executeFromState(ConfigNodeProcedureEnv env, AddRegionPeerState state)
+      throws ProcedureSuspendedException, ProcedureYieldException, InterruptedException {
+    if (consensusGroupId == null) {
+      return Flow.NO_MORE_STATE;
+    }
+    RegionMaintainHandler handler = env.getRegionMaintainHandler();
+    try {
+      outerSwitch:
+      switch (state) {
+        case CREATE_NEW_REGION_PEER:
+          handler.createNewRegionPeer(consensusGroupId, destDataNode);
+          setNextState(AddRegionPeerState.DO_ADD_REGION_PEER);
+          break;
+        case DO_ADD_REGION_PEER:
+          TSStatus tsStatus =
+              handler.addRegionPeer(this.getProcId(), destDataNode, consensusGroupId, coordinator);
+          TRegionMigrateResult result;
+          if (tsStatus.getCode() == SUCCESS_STATUS.getStatusCode()) {
+            result = handler.waitTaskFinish(this.getProcId(), coordinator);
+          } else {
+            throw new ProcedureException("ADD_REGION_PEER executed failed in DataNode");
+          }
+          switch (result.getTaskStatus()) {
+            case TASK_NOT_EXIST:
+              // coordinator crashed and lost its task table
+            case FAIL:
+              // maybe some DataNode crash
+              LOGGER.warn(
+                  "result is {}, will use resetPeerList to clean in the future",
+                  result.getTaskStatus());
+              //              List<TDataNodeLocation> correctDataNodeLocations =
+              //
+              // env.getConfigManager().getPartitionManager().getAllReplicaSets().stream()
+              //                      .filter(
+              //                          tRegionReplicaSet ->
+              //
+              // tRegionReplicaSet.getRegionId().equals(consensusGroupId))
+              //                      .findAny()
+              //                      .get()
+              //                      .getDataNodeLocations();
+              //              handler.resetPeerList(consensusGroupId, correctDataNodeLocations);
```

Comment:
Why left those comment lines?

Reply:
Unfinished functionality, will finish in a few weeks...